packages/alfa-rules/src/sia-r8/rule.ts

```diff
@@ -65,11 +82,17 @@ export default Rule.Atomic.of<Page, Element>({
  * @public
  */
 export namespace Outcomes {
-  export const HasName = (role: Role.Name) =>
+  export const FormFieldWithAriaRoleHasName = (role: Role.Name) =>
     Ok.of(WithRole.of(`The form field has an accessible name`, role));
 
-  export const HasNoName = (role: Role.Name) =>
+  export const FormFieldWithAriaRoleHasNoName = (role: Role.Name) =>
     Err.of(
       WithRole.of(`The form field does not have an accessible name`, role),
     );
+
+  export const InputElementWithNoAriaRoleHasName = (typeAttribValue: string) =>
+    Ok.of(Diagnostic.of(`The type="${typeAttribValue}" form field has an accessible name`));
+
+  export const InputElementWithNoAriaRoleHasNoName = (typeAttribValue: string) =>
+    Err.of(Diagnostic.of(`The type="${typeAttribValue}" form field does not have an accessible name`));
```

Comment:
**Suggestion:** we should tighten the types by using the same `InputType` type that `hasInputType` uses (instead of `string`). It is currently not exported, so it should ideally be re-exported from `alfa-dom/Element` 🤔